### PR TITLE
Remaining climate setting schedule hooks

### PR DESCRIPTION
**Pull request by Andrewjeska**

### CODE REVIEW

File: src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+} from '@tanstack/react-query'
+import type { ClimateSettingScheduleDeleteRequest, SeamError } from 'seamapi'
+
+import { NullSeamClientError, useSeamClient } from 'lib/seam/use-seam-client.js'
+
+export type UseDeleteClimateSettingScheduleMutationParams =
+  ClimateSettingScheduleDeleteRequest
+
+export function useDeleteClimateSettingSchedule(): UseMutationResult<
+  unknown,
```
Comment (Andrewjeska) [inline src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts:14]: The endpoint returns `{}`, is `unknown` the recommended return type here?

Comment (razor-x) [reply src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts:14]: How about via https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492
```suggestion
  Record<string, never>,
```

**Review by razor-x — APPROVED** [review]

File: src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts (hunk 1 above)
Comment (Andrewjeska) [reply src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts]: Just kidding the sdk returns `{ok: true}`, I was looking at connect

https://github.com/seamapi/react/pull/364/commits/6f9e86c93f0bca4ea30b137122b4cff1a723dfee

File: src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts [hunk 2]
```diff
@@ -3,16 +3,19 @@ import {
...
 import { NullSeamClientError, useSeamClient } from 'lib/seam/use-seam-client.js'
 
 export type UseDeleteClimateSettingScheduleMutationParams =
   ClimateSettingScheduleDeleteRequest
 
-export interface UseDeleteClimateSettingScheduleData {
-  ok: true
-}
+export type UseDeleteClimateSettingScheduleData =
+  ClimateSettingScheduleDeleteResponse
```
Comment (razor-x) [inline src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts:18]: It looks like in the other hooks we explicitly define the type here and do not use `*Response`. I believe the reasoning for this is better editor / type dx. I think we should keep this pattern consistent in the code base, but it's something we might be able to improve on with `@seamapi/types`

```suggestion
export type UseDeleteClimateSettingScheduleData = void
```

**Review by razor-x — APPROVED** [review]

File: src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts (hunk 2 above)
Comment (Andrewjeska) [reply src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts]: Sorry, I'm getting
```
/home/runner/work/react/react/src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts
Error:   13:51  error  void is only valid as a return type or generic type argument  @typescript-eslint/no-invalid-void-type

```

Can we keep it as it was before?

Comment (Andrewjeska) [reply src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts:18]: I get the same thing if I pass in `void` to `UseMutationResult<>`

Comment (razor-x) [reply src/lib/seam/thermostats/climate-setting-schedules/use-delete-climate-setting-schedule.ts:18]: Sure. This is annoying but I get it.

**Pull request merged**